File: app/src/main/java/com/faendir/lightning_launcher/multitool/proxy/Scriptable.java
package com.faendir.lightning_launcher.multitool.proxy;

/**
 * @author lukas
 * @since 17.07.18
 */
public interface Scriptable extends Proxy {
    Object get(String key, Scriptable start);

    void put(String key, Scriptable start, Object value);
}

File: app/src/main/java/com/faendir/lightning_launcher/multitool/proxy/Scriptable.kt
package com.faendir.lightning_launcher.multitool.proxy

/**
 * @author lukas
 * @since 17.07.18
 */
interface Scriptable : Proxy {
    operator fun get(key: String, start: Scriptable): Any

    fun put(key: String, start: Scriptable, value: Any)
}
